### PR TITLE
fix: broken ProfileEditPage route

### LINKED ISSUE #9574: ProfileEditPage route is broken
## Observed behavior

When viewing your profile, if you click "Edit" the console shows the error that the ShowEditProfile route does not exist.

There is a route defined - I'm suspecting something is just wired up wrongly somewhere but I'm not seeing where at a quick glance.

## Expected behavior

Users should be able to edit their profile.

## User-facing consequences

User couldn't edit their profile.

## Steps to reproduce

Start devserver, sign in, try to edit your profile.

## Context

0.16
Fedora 36
Chromium

### PR DESCRIPTION
**Pull request by camperjett**
Signed-off-by: Jaideep Sharma sharmajaideep1906@gmail.com
## Summary
* Previously Editing Profile did not work.
* Here the Issue was that constants used to map component names had wrong values hence they were linking to wrong components. 
* I changed the constant values to correct component names.

## References
Fixes #9574 

## Reviewer guidance

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by MisRob** [comment]
@camperjett Thank you, all checks have passed which is great. I will review the code changes next week. Meanwhile, if you'd also like to add yourself to [Contributors list](https://kolibri-dev.readthedocs.io/en/develop/contributing/authors.html), please push a commit to this PR with updated https://github.com/learningequality/kolibri/blob/develop/AUTHORS.md

**Comment by marcellamaki** [comment]
Thank you @camperjett!

**Comment by pcenov** [comment]
Manual QA passes as well - clicking the 'Edit' button does bring the user to /en/profile/#/edit, the changes made are saved correctly and the user is brought back to /en/profile/#/

**Comment by akolson** [comment]
@camperjett Thank you!

File: kolibri/plugins/user_profile/assets/src/constants.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export const ComponentMap = {
-  PROFILE: 'ProfilePage',
-  PROFILE_EDIT: 'ProfileEditPage',
+  PROFILE: 'PROFILE',
+  PROFILE_EDIT: 'PROFILE_EDIT',
   NEW_PASSWORD: 'NewPasswordPage',
```
Comment (jredrejo) [inline kolibri/plugins/user_profile/assets/src/constants.js]: this should be deleted as it's not in use anymore.
Also, it would not block this PR from being merged, but it would be a good idea to refactor the name `ComponentMap` to `RoutesMap`  or similar as these are not component names anymore

**Review by jredrejo — CHANGES_REQUESTED** [review]
It's fine, just a small change required to remove legacy code

File: kolibri/plugins/user_profile/assets/src/constants.js (hunk 1 above)
Comment (camperjett) [reply kolibri/plugins/user_profile/assets/src/constants.js]: Are you referring to line 4? In that case this constant is being used in some of the files.

Comment (jredrejo) [reply kolibri/plugins/user_profile/assets/src/constants.js]: Yes, line 4, I don't see `NEW_PASSWORD` in use in this plugin, and the `NewPasswordPage` component does not exist anymore, it was replaced by a modal window. Where do you see it's being used?

Comment (MisRob) [reply kolibri/plugins/user_profile/assets/src/constants.js]: Thank you, @jredrejo.

@camperjett Some of our routes don't have `name` and only have `component` as an identifier. For such routes, we usually create these maps so that we don't need to reference components' names directly from `$router.getRoute`. However, `PROFILE` and `PROFILE_EDIT` routes already have names. Therefore we don't need to use this map anymore at all (no need to map a route name to the same route name).

To sum it up together with @jredrejo's suggestion, I'd suggest removing this map completely and then simply using  `$router.getRoute('PROFILE_EDIT')`, for example.

Comment (MisRob) [reply kolibri/plugins/user_profile/assets/src/constants.js]: @camperjett @jredrejo Neither comment is blocking though I'd say. This fixes the bug, and our feedback is related to some cleanup of legacy routing things.

@camperjett You're welcome to push those changes if you'd like, leaving it open for some time. Thanks for cooperation.

**Review by radinamatic — APPROVED** [review]
Tested by @pcenov, good to go! 💯 :shipit:

File: kolibri/plugins/user_profile/assets/src/constants.js (hunk 1 above)
Comment (camperjett) [reply kolibri/plugins/user_profile/assets/src/constants.js]: PTAL

**Review by jredrejo — APPROVED** [review]
Thanks @camperjett
